### DISCUSSION TITLE
Question regarding database shema person - relationsships (many to many)

### DISCUSSION THREAD

**Discussion by andreas83** [prisma/prisma#24392] [Q&A] [answered]
### Question

i have two tables **person** and **relation** and need a way to store the relationships between persons. 
i.e. person a. is mother of person b, person c. is sister of person d. etc. 

this is what i tried so far:

```prisma
model Person {
  id        Int      @id @default(autoincrement())
  createdAt DateTime @default(now())
  updatedAt DateTime @updatedAt

  titel        String?
  firstname      String?
  lastname     String?


  fromRelation Relation[] @relation("PersonInRelationA", fields: [id], references: [personAId])
  toRelation Relation[] @relation("PersonInRelationB", fields: [id], references: [personBId])
}

model Relation {
  id          Int      @id @default(autoincrement())

  typ         String   

  personA     Person?   @relation("PersonInRelationA", fields: [personAId], references: [id])
  personAId   Int?
  personB     Person?   @relation("PersonInRelationB", fields: [personBId], references: [id])
  personBId   Int?


}
```

The error i encounter is:

`error: Error parsing attribute "@relation": The relation field `fromRelation` on Model `Person` must not specify the `fields` or `references` argument in the @relation attribute. You must only specify it on the opposite field `personA` on model `Relation`.`

But when i remove fromRelation and to Relation from Person i get: 

Error validating field `personB` in model `Relation`: The relation field `personB` on model `Relation` is missing an opposite relation field on the model `Person`. Either run `prisma format` or add it manually.

So i am kinda stuck here, help would be highly appreciated. 

thank in advanced 

andreas

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by ludralph** [comment, marked as answer]
Hi @andreas83 👋 

Thank you for raising this question. You can modify your schema as shown below:
```prisma
model Person {
  id        Int      @id @default(autoincrement())
  createdAt DateTime @default(now())
  updatedAt DateTime @updatedAt

  titel        String?
  firstname      String?
  lastname     String?


  fromRelation Relation[] @relation("PersonInRelationA")
  toRelation Relation[] @relation("PersonInRelationB")
}

model Relation {
  id          Int      @id @default(autoincrement())

  typ         String   

  personA     Person?   @relation("PersonInRelationA", fields: [personAId], references: [id])
  personAId   Int?
  personB     Person?   @relation("PersonInRelationB", fields: [personBId], references: [id])
  personBId   Int?


}
```

**Reply by andreas83** [reply]
thank you :)

**Comment by anasabdulla1204-ux** [comment]
model Person {
  id        Int      @id @default(autoincrement())
  createdAt DateTime @default(now())
  updatedAt DateTime @updatedAt

  titel        String?
  firstname      String?
  lastname     String?


  fromRelation Relation[] @relation("PersonInRelationA")
  toRelation Relation[] @relation("PersonInRelationB")
}

model Relation {
  id          Int      @id @default(autoincrement())

  typ         String   

  personA     Person?   @relation("PersonInRelationA", fields: [personAId], references: [id])
  personAId   Int?
  personB     Person?   @relation("PersonInRelationB", fields: [personBId], references: [id])
  personBId   Int?


}
